Mathlib/Data/ZMod/Module.lean

```diff
@@ -0,0 +1,41 @@
+import Mathlib.Data.ZMod.Basic
+import Mathlib.Algebra.Module.LinearMap
+
+/-!
+# The `ZMod n`-module structure on commutative monoids whose elements have order dividing `n`
+-/
+
+variable {n : ℕ} {M M₁ : Type*} [AddCommMonoid M] [AddCommMonoid M₁]
+  [Module (ZMod (n + 1)) M] [Module (ZMod (n + 1)) M₁]
```

Comment:
Why not `ZMod n`? `ZMod 0` is defeq to `Z` so `map_smul` should just follow from `map_zsmul`.

Reply:
Done, I just had to take an `AddCommGroup` instance instead of `AddCommMonoid`, although I'm not sure how much that matters in practice since I don't think something can be a Z/nZ-module without being a group

Reply:
Yeah, every `ZMod n` is a ring, so any AddCommMonoid that is a `ZMod n`-module must be a group (define `-a = (-1) • a`). In mathlib we almost always write AddCommGroup when it's a module over a Ring (rather than Semiring), so changing to AddCommGroup conforms with mathlib convention. However, if you do want to prove the AddCommMonoid version, you can do so by temporarily constructing a AddCommGroup within the proof.